### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by maelle**
Next time I should try to use gitflow. 👼 

* I am not able yet to reproduce the error from CRAN 😢 In the latest Travis build the failing job is the one on R devel on OSX... https://travis-ci.org/maelle/HIBPwned/builds/356264582

* I have added some trying and waiting in utils.R

* I have added grouping to the reference in the pkgdown website

* I have taken the freedom to use `styler` and `lintr` to mostly shorten lines, add space, and rename "URLS" to "urls" (I added exceptions for non snake case function names)

### CODE REVIEW

**Comment by maelle** [comment]
I could tacke #5 (breaking change though...) and #6 if you agree with them.

**Comment by stephlocke** [comment]
I keep URLS uppercase as it avoids naming collisions with some functions that exist in different packages. I'd prefer it to stay as is if that's ok.

**Comment by maelle** [comment]
This could be merged but I still get a failing build on OSX R devel (see [this log for instance](https://travis-ci.org/maelle/HIBPwned/jobs/358299812)) because of missing dependencies, and I still wasn't able to reproduce the CRAN error.

**Comment by maelle** [comment]
@stephlocke 

* Can I merge this? The [failure on Mac R devel](https://travis-ci.org/stephlocke/HIBPwned/jobs/358321151#L1204) seems to be a Travis problem, not a R problem.

* Does the package need anything more before the next CRAN release? this PR closes all 3 open issues, and the CRAN note will disappear.

**Comment by stephlocke** [comment]
Travis fail? https://travis-ci.org/stephlocke/HIBPwned/jobs/358321151
Is it possible to write a test for the get and retry funciton you made?

**Comment by maelle** [comment]
the Travis fail seems to be a Travis issue (it fails to install dependencies on Mac R devel). I could investigate it or remove the Mac R devel build from the matrix?

**Comment by stephlocke** [comment]
Yeah, I don't really care all that much about mac ;)

File: DESCRIPTION [hunk 1]
```diff
@@ -8,24 +8,28 @@ Authors@R: c(person("Steph", "Locke"
...
     knitr,
-    rmarkdown
+    rmarkdown,
+    lintr
 Imports:
     httr,
     urltools,
     jsonlite,
     ratelimitr
-RoxygenNote: 6.0.1
```
Comment (stephlocke) [inline DESCRIPTION:28]: Any reason why you've incremented for the dev version here?

File: DESCRIPTION [hunk 2]
```diff
@@ -8,24 +8,28 @@ Authors@R: c(person("Steph", "Locke"
...
+RoxygenNote: 6.0.1.9000
 VignetteBuilder: knitr
-URL: https://github.com/stephlocke/HIBPwned
```
Comment (stephlocke) [inline DESCRIPTION:30]: Maybe better to just refer to the pkgdown site

**Review by stephlocke — COMMENTED** [review]
Couple of minor questions on this.

File: DESCRIPTION (hunk 2 above)
Comment (maelle) [reply DESCRIPTION:30]: One can do both, and as a potential contributor and stargazer I prefer seeing the repo ☺

File: DESCRIPTION (hunk 1 above)
Comment (maelle) [reply DESCRIPTION:28]: No I should change this

Comment (stephlocke) [reply DESCRIPTION:28]: 👍

File: DESCRIPTION (hunk 2 above)
Comment (stephlocke) [reply DESCRIPTION:30]: But we have the github in the bugs field, but meh, I don't have particularly strong feelings about this so sure thing, we'll go with both

File: DESCRIPTION (hunk 1 above)
Comment (maelle) [reply DESCRIPTION:28]: Now I know I have a reason 😿

**Review by stephlocke — APPROVED** [review]

**Pull request merged**